File: OPUS-tauri-app/src/types/diagnostics.ts
export type EnvSnapshot = {
  path: string;
  shell: string;
  home: string;
  ssh_auth_sock?: string;
  git_ssh_command?: string;
  git_askpass?: string;
  gh_config_dir?: string;
  gh_auth_summary?: string;
};

export type SessiondStatus = {
  running: boolean;
  version?: string;
  socket_path?: string;
  last_error?: string;
  last_restart?: string;
};

export type CliStatus = {
  available: boolean;
  path?: string;
  version?: string;
  error?: string;
};

File: OPUS-tauri-app/src-tauri/src/types/diagnostics.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct EnvSnapshot {
    pub path: String,
    pub shell: String,
    pub home: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub ssh_auth_sock: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub git_ssh_command: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub git_askpass: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub gh_config_dir: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub gh_auth_summary: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct SessiondStatus {
    pub running: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub version: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub socket_path: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_error: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_restart: Option<String>,
}
